tutorials/scripts/02_Intro_to_DBO/README.rst

```diff
@@ -0,0 +1,84 @@
+
+First we have to define the Hyperparameter search space as well as the run function, which, given a certain ``config`` of hyperparameters, should return the objective we want to maximize. We are computing the 10-D (:math:`d = 10`) Ackley function with :math:`a = 20`, :math:`b = 0.2` and :math:`c = 2\pi` and want to find its minimum (:math:`f(x=(0, \dots , 0)) = 0`) on the domain :math:`[-32.768, 32.768]^10`.
+Thus we define the hyperparameter problem as :math:`x_i \in [-32.768, 32.768] \forall i \in 1..10` and the objective returned by the ``run`` function as :math:`-f(x)`.
+
+.. literalinclude:: ackley.py
+   :language: python
+   :caption: **file**: ``ackley.py``
+
+Definition of the distributed Bayesian optimization search (DBO)
+----------------------------------------------------------------
+
+.. image:: figures/cbo_vs_dbo.jpg
+   :scale: 100 %
+   :alt: CBO - DBO architecture comparison
+   :align: center
+
+DBO (*b*) is very similar to CBO (Centralized Bayesian Optimization) (*a*) in the sense that we iteratively generate new configurations with an optimizer (*O*), evaluate them on Workers (*W*) (calling :math:`f`, which takes :math:`t_{eff}`), and fit the optimizer on the history of the search (the configuration/objective pairs) to generate better configurations.
+The only difference is that with CBO the fitting of the optimizer and generation of new configurations is centralized on a Manager *M*, while with DBO each worker has its own optimizer and these operations are parallelized.
+This difference makes DBO a preferable choice when the run function is too quick and the number of workers too big ; with a large enough number of workers the fit of the optimizer (which has to be performed each time we generate a configuration) starts to take more time than the run function takes to be evaluated : at that point we obtain a congestion on the manager and workers out of work waiting for a configuration to evaluate. DBO can manage these numbers of workers by parallelizing the whole process.
+
+DBO may be described in the following algorithm:
+
+.. image:: figures/dbo_algo.jpg
+   :scale: 100 %
+   :alt: DBO algorithm
+   :align: center
+
+Setup DBO
+_________
+
+Unlike CBO, DBO doesn't use any evaluator instance, everything is comprised within the DBO search instance :
+
+.. code-block:: python
+   :caption: **file**: ``DBO.py``
+
+   from ackley import hp_problem, run
+
+   search = DBO(
+      hp_problem,
+      run,
+   )
+
+Execution of the search : using MPI
+-----------------------------------
+
+The backend of DBO uses MPI, so if we want for example to get the final ``results`` and save it at a specific location, we need to get it from a single worker while executing the search for the same amount of time on each one of them :
```

Comment:
put a note about `checkpoint_file` and `checkpoint_freq` mention sync and async parameters and put a note as well